workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Readers/BlittableReaderWriterBase.cs

```diff
@@ -28,5 +28,21 @@ public sealed override TSpatialComponentData Data
                 }
             }
         }
+
+        public override void Send(TComponentUpdate update)
+        {
+            try
+            {
+                var data = EntityManager.GetComponentData<TSpatialComponentData>(Entity);
+                UpdateData(ref data, update);
+                EntityManager.SetComponentData(Entity, data);
+            }
+            catch (Exception e)
+            {
+                throw new WriterDataUpdateFailedException(e);
```

Comment:
Please pass in the entity index here and in other exceptions

Reply:
Any particular reason for the entity index and not the spatial ID? That's what we've been logging in most other exceptions I think, and if an entity had writers injected we know it must have a spatial ID

Reply:
Actually accessing the spatial ID is not something we can do in this context without extra injection, so I'll just do it with the entity ID (probably not worth the trouble of making it nicer for an exception that should never happen)